app/main.py

```diff
@@ -1,8 +1,44 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(
+            self,
+            name: str,
+            age: int,
+
+    ) -> None:
+        self.name = name
+        self.age = age
+        self.wife = None
+        self.husband = None
+        Person.people[self.name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    list_of_humans = []
+    for person in people:
+        wife = None
+        husband = None
+        if "wife" in person:
+            wife = person["wife"]
+        elif "husband" in person:
+            husband = person["husband"]
+        mate = Person(person["name"], person["age"])
+        mate.wife = wife
+        mate.husband = husband
+        list_of_humans.append(mate)
+
+    for person in Person.people:
+        if Person.people[person].wife:
+            partner_name = Person.people[person].wife
+            Person.people[person].wife = Person.people[partner_name]
+            del Person.people[person].husband
+        elif Person.people[person].husband:
+            partner_name = Person.people[person].husband
+            Person.people[person].husband = Person.people[partner_name]
+            del Person.people[person].wife
+        else:
+            del Person.people[person].wife
+            del Person.people[person].husband
+
+    return list_of_humans
```

Comment:
```suggestion
for spouse in people:
    person = Person.people[spouse["name"]]
    
    if wife := spouse.get("wife"):
        person.wife = ...
    elif husband := spouse.get("husband")
        person.husband = ...
```
You can use the walrus operator here; it will make your code cleaner and